vllm/engine/llm_engine.py

```diff
@@ -1938,6 +1939,18 @@ def _validate_model_inputs(self, inputs: Union[DecoderOnlyInputs,
         if prompt_ids is None or len(prompt_ids) == 0:
             raise ValueError("Prompt cannot be empty")
 
+        # Guard against out-of-vocab tokens.
+        # For some tokenizers, tokenizer.decode will happily return empty text
+        # for token ids that are out of vocab, and we don't detect token ids
+        # that are greater than the max token id before running the model.
+        # However, these token ids will later crash a cuda kernel at runtime
+        # with an index out of bounds error. This will crash the entire engine.
+        tokenizer = self.get_tokenizer(lora_request)
```

Comment:
@joerunde I think we may need to also handle the case where there is no tokenizer (i.e. if the engine is run with `skip_tokenizer_init`). Unfortunately the `get_tokenizer` method will raise an exception in this case, might have to just wrap this in `if self.tokenizer_group is not None`.

Reply:
This would hopefully also have been caught by https://github.com/vllm-project/vllm/blob/main/tests/engine/test_skip_tokenizer_init.py :)

Reply:
Ah, nice catch!
What is `--skip-tokenizer-init` used for?

Reply:
Some people use vLLM to deploy models without any tokenizer so it's completely token ids in and out.